### PR TITLE
Replace the old OTP modal with a new Airship version

### PR DESCRIPTION
**Pull request by swansontec**
This one is for @amanalili to review, test, and, fix. It's basically combines the art/walletList-modalRefactor-otpResetModal branch with work from one of my old cleanup branches.

### CODE REVIEW

File: src/components/scenes/WalletListScene.js [hunk 1]
```diff
@@ -10,7 +10,6 @@ import AntDesignIcon from 'react-native-vector-icons/AntDesign'
 import Ionicon from 'react-native-vector-icons/Ionicons'
 
 import credLogo from '../../assets/images/cred_logo.png'
-import iconImage from '../../assets/images/otp/OTP-badge_sm.png'
```
Comment (swansontec) [inline src/components/scenes/WalletListScene.js:13]: Clearly, this line needs to come back using @amanalili's new `ModalImageIcon` component.

**Comment by swansontec** [comment]
I combined this with Art's #1883.

By our powers combined, I am captain ~~planet~~ pull request.

File: src/components/scenes/WalletListScene.js [hunk 2]
```diff
@@ -73,29 +72,31 @@ type DispatchProps = {
...
+        <TwoButtonSimpleConfirmationModal
+          bridge={bridge}
+          icon={<Image source={otpIcon} />}
+          title={s.strings.otp_modal_reset_headline}
+          subTitle={s.strings.otp_modal_reset_description}
+          cancelText={s.strings.request_review_answer_no}
+          doneText={s.strings.request_review_answer_yes}
+        />
+      ))
+      resolved ? this.props.keepOtp() : this.props.disableOtp()
```
Comment (amanalili) [inline src/components/scenes/WalletListScene.js:99]: We could separate the modal creation into a separate function? So if ever we have other logic on componentDidMount it won't be cluttered.

Comment (swansontec) [reply src/components/scenes/WalletListScene.js:99]: I like that idea. Let's do it.

**Pull request merged**